### PR TITLE
Intelligently add parens when inlining local varaibles

### PR DESCRIPTION
**Pull request by gfreezy**
fixed this. https://github.com/rust-analyzer/rust-analyzer/pull/1037#discussion_r268627141

### CODE REVIEW

File: crates/ra_syntax/src/grammar.ron [hunk 1]
```diff
@@ -582,6 +582,7 @@ Grammar(
             options: [
                 ["pat", "Pat"],
                 ["initializer", "Expr"],
+                ["initializer_macro", "MacroCall"],
```
Comment (matklad) [inline crates/ra_syntax/src/grammar.ron]: Why do we need this accessor? macro should be treated as an expression in this position. Perhaps there's a bug in the parser?

Comment (gfreezy) [reply crates/ra_syntax/src/grammar.ron]: macro is not an expr😖

**Comment by matklad** [comment]
It should be! I think just adding it to the expr enum  in grammar.ron
should do the trick?

**Comment by gfreezy** [comment]
```
--- a/crates/ra_syntax/src/grammar.ron
+++ b/crates/ra_syntax/src/grammar.ron
@@ -494,6 +494,7 @@ Grammar(
                 "RangeExpr",
                 "BinExpr",
                 "Literal",
+                "MacroCall",
             ],
         ),

@@ -544,7 +545,7 @@ Grammar(
         "NameRef": (),
         "MacroCall": (
             traits: [ "NameOwner", "AttrsOwner" ],
-            options: [ "TokenTree", "Path" ],
+            options: [ "TokenTree", "Path", "Expr" ],
         ),
         "Attr": ( options: [ ["value", "TokenTree"] ] ),
         "TokenTree": (),
```

**Comment by gfreezy** [comment]
rebased.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**